src/index.js

```diff
@@ -50,7 +51,7 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // Avoid dispatching an action if the store is already up-to-date,
     // even if `history` wouldn't do anything if the location is the same
     if(getRouterState().path !== locationToString(location)) {
-      store.dispatch(updatePath(locationToString(location)));
+      store.dispatch(updatePath(locationToString(location), true));
```

Comment:
You are actually right, it should't change anything because of the check `currentState != currentLocation`, that `true` doesn't add anything in terms of functionality. I added it for clarity, to make it clear that the action is not meant to update the route, but just the state. I'll remove that `true` if you think is redundant.